src/filters/arrays.js

```diff
@@ -50,8 +51,9 @@ function matchItems(array1, array2, index1, index2, context) {
   }
   var objectHash = context.objectHash;
   if (!objectHash) {
-    // no way to match objects was provided, try match by position
-    return context.matchByPosition && index1 === index2;
+    var ivalue1 = Immutable.Map(value1);
+    var ivalue2 = Immutable.Map(value2);
+    return Immutable.is(ivalue1, ivalue2);
```

Comment:
again, regardless of performance:

> I don't think a deep equals is a good generic solution to this problem, it seems to me defaulting to same-position is still a reasonable solution for most cases, because as you say, knowing if something has changed or moved can't be solved trivially, so to optimize those cases, if it is possible, I think is better to expect objectHash option to be provided by the user.

which is to me the best solution for this case, you can just specify an objectHash function using Immutable and that's perfect. I appreciate the contribution (I didn't know about Immutable), but I still think the default behavior on this situation shouldn't be a deep compare.